src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +86,190 @@ void VulnerabilityScannerFacade::start(
+                    {
+                        std::shared_ptr<std::string> returnStr = m_syncQueue->pop();
+
+                        flatbuffers::Verifier syncVerifier(reinterpret_cast<const uint8_t*>(returnStr->c_str()),
+                                                           returnStr->size());
+
+                        if (SyscollectorSynchronization::VerifySyncMsgBuffer(syncVerifier))
+                        {
+                            const SyscollectorSynchronization::SyncMsg* syncMsg =
+                                SyscollectorSynchronization::GetSyncMsg(
+                                    reinterpret_cast<const char*>(returnStr->c_str()));
+                            std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*>
+                                data = syncMsg;
+
+                            std::string packageName;
+                            if (syncMsg->data_as_state() &&
+                                syncMsg->data_as_state()->attributes_as_syscollector_packages() &&
+                                syncMsg->data_as_state()->attributes_as_syscollector_packages()->name())
+                            {
+                                packageName =
+                                    syncMsg->data_as_state()->attributes_as_syscollector_packages()->name()->c_str();
+                            }
+
+                            try
+                            {
+                                logDebug1(
+                                    WM_VULNSCAN_LOGTAG, "Retry syscollector_packages event: %s package", packageName);
+                                scanOrchestrator->run(data);
+                            }
+                            catch (const std::exception& e)
+                            {
+                                logWarn(WM_VULNSCAN_LOGTAG,
+                                        "Discarded syscollector_packages event: %s package. %s",
+                                        packageName,
+                                        e.what());
+                            }
+                        }
+                    }
+                }
+            });
+
         // Subscription to syscollector delta events.
         m_syscollectorDeltasSubscription =
             std::make_unique<RouterSubscriber>("deltas-syscollector", "vulnerability_scanner_deltas");
         m_syscollectorDeltasSubscription->subscribe(
-            [scanOrchestrator](const std::vector<char>& message)
+            [scanOrchestrator, this](const std::vector<char>& message)
             {
+                const SyscollectorDeltas::Delta* delta = SyscollectorDeltas::GetDelta(message.data());
                 std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*> data =
-                    SyscollectorDeltas::GetDelta(message.data());
+                    delta;
+
+                std::string packageName;
```

Comment:
Missing initialization
 postponed: ���c�^? package. Queue size: 1473

```suggestion
                std::string packageName {};
```
or
```suggestion
                std::string packageName {"undefined"};
```

applies to the other cases



Sorry for the back and forth, I thought the ternary operation you used was okay. I've just thought that having all of that on the log function was too crumbled togheter

Reply:
The std::string default constructor is an empty string, the issue was that I was using std::string for a function that expects char pointer.

Reply:
Perfect